extensions/iceberg/src/main/java/io/deephaven/iceberg/util/SortOrderProvider.java

```diff
@@ -0,0 +1,33 @@
+//
+// Copyright (c) 2016-2025 Deephaven Data Labs and Patent Pending
+//
+package io.deephaven.iceberg.util;
+
+/**
+ * A specification for providing {@link org.apache.iceberg.SortOrder} while writing to an iceberg table.
+ */
+public interface SortOrderProvider {
+
+    // Static factory methods for creating SortOrderProvider instances
+
+    /**
+     * Do not sort the data while writing new data to the iceberg table.
+     */
+    static SortOrderProvider disableSorting() {
+        return SortOrderProviderInternal.DisableSorting.getInstance();
+    }
+
+    /**
+     * Use the default {@link org.apache.iceberg.Table#sortOrder()} of the table while writing new data.
```

Comment:
Probably worth mentioning that it's the sort order of the Table at the time the writer is constructed (ie, it's not dynamic). We should probably mention that if no sort order is set on the table, no sorting will be done.